src/main/kotlin/com/openlattice/mechanic/upgrades/MigrateOrgPermissionsUpgrade.kt

```diff
@@ -0,0 +1,104 @@
+package com.openlattice.mechanic.upgrades
+
+import com.hazelcast.query.Predicates
+import com.openlattice.authorization.Ace
+import com.openlattice.authorization.AceKey
+import com.openlattice.authorization.AceValue
+import com.openlattice.authorization.Acl
+import com.openlattice.authorization.Action
+import com.openlattice.authorization.mapstores.PermissionMapstore
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.external.ExternalDatabaseConnectionManager
+import com.openlattice.postgres.external.ExternalDatabasePermissioningService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.sql.SQLException
+import java.util.UUID
+
+class MigrateOrgPermissionsUpgrade(
+        toolbox: Toolbox,
+        private val exConnMan: ExternalDatabaseConnectionManager,
+        private val exDbPermMan: ExternalDatabasePermissioningService
+): Upgrade {
+
+    val logger: Logger = LoggerFactory.getLogger(MigrateOrgPermissionsUpgrade::class.java)
+
+    private val externalColumns = HazelcastMap.EXTERNAL_COLUMNS.getMap(toolbox.hazelcast)
+    private val permissions = HazelcastMap.PERMISSIONS.getMap(toolbox.hazelcast)
+
+    override fun upgrade(): Boolean {
+        // filtering for a specific org
+        println("Organization to filter: ")
+        val input = readLine()?.ifBlank { null }
+        val filterFlag = (input != null)
+        val filteringOrgID = if (filterFlag) {
+            UUID.fromString(input!!)
+        } else {
+            UUID(0, 0)
+        }
+
+        if (filterFlag) {
+            logger.info("Filtering for org {}", filteringOrgID)
+        } else {
+            logger.info("No filtering")
+        }
+
+        val filteredPermissionEntries = permissions.entrySet(
+                Predicates.`in`<AceKey, AceValue>(PermissionMapstore.SECURABLE_OBJECT_TYPE_INDEX,
+                        SecurableObjectType.PropertyTypeInEntitySet,
+                        SecurableObjectType.OrganizationExternalDatabaseColumn
+                )
+        ).filter { entry ->
+            try {
+                val columnId = entry.key.aclKey[1]
+                val column = externalColumns[columnId]
+                return !filterFlag || column?.organizationId == filteringOrgID
```

Comment:
@UnsungHero97 I tested this a little, and I think this will help: 
```suggestion
                return@filter !filterFlag || column!!.organizationId == filteringOrgID
```
That way, the null exception will actually happen, and the `catch` will actually get triggered. 

I think what we seen during migration is that the `return` returns false entirely (not just for `filter`!). Also, I do think you need `!!` here so that null errors won't silently fail (won't get caught by the exception handler).